src/Libraries/Microsoft.Extensions.AI.Abstractions/SpeechToText/SpeechToTextResponseUpdateKind.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Describes the intended purpose of a specific update during streaming of speech to text updates.
+/// </summary>
+[JsonConverter(typeof(Converter))]
+public readonly struct SpeechToTextResponseUpdateKind : IEquatable<SpeechToTextResponseUpdateKind>
+{
+    /// <summary>Gets when the generated text session is opened.</summary>
+    public static SpeechToTextResponseUpdateKind SessionOpen { get; } = new("sessionopen");
+
+    /// <summary>Gets when a non-blocking error occurs during speech to text updates.</summary>
+    public static SpeechToTextResponseUpdateKind Error { get; } = new("error");
+
+    /// <summary>Gets when the text update is in progress, without waiting for silence.</summary>
+    public static SpeechToTextResponseUpdateKind TextUpdating { get; } = new("textupdating");
+
+    /// <summary>Gets when the text was generated after small period of silence.</summary>
+    public static SpeechToTextResponseUpdateKind TextUpdated { get; } = new("textupdated");
+
+    /// <summary>Gets when the generated text session is closed.</summary>
+    public static SpeechToTextResponseUpdateKind SessionClose { get; } = new("sessionclose");
```

Comment:
Is the expectation that in an update sequence you always get a session open, then zero or more pairs of textupdating/textupdated, and then a session close, with zero or more errors sprinkled throughout?

Reply:
Yes, except for Whisper.net streaming API's all other cloud API's had the concept of Session being started and ended. I implemented with ease this concept using the Whisper.net, as a good standard.